test-case/multiple-pause-resume.sh

```diff
@@ -0,0 +1,212 @@
+    local rate=$(func_pipeline_parse_value $idx rate)
+    local fmt=$(func_pipeline_parse_value $idx fmt)
+    local dev=$(func_pipeline_parse_value $idx dev)
+    local pcm=$(func_pipeline_parse_value $idx pcm)
+    local type=$(func_pipeline_parse_value $idx type)
+    # expect is tcl language script
+    #   expr rand(): produces random numbers between 0 and 1
+    #   after ms: Ms must be an integer giving a time in milliseconds.
+    #       The command sleeps for ms milliseconds and then returns.
+    dlogi "$pcm to command: $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file -q"
+    expect <<END &
+spawn $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file -q
+set i 1
+expect {
+    "*#*+*\%" {
+        set sleep_t [expr int([expr rand() * $rnd_range]) + $rnd_min ]
+        puts "\r(\$i/$repeat_count) $pcm $idx: Wait for \$sleep_t ms before pause"
+        send " "
+        after \$sleep_t
+        exp_continue
+    }
+    "*PAUSE*" {
+        set sleep_t [expr int([expr rand() * $rnd_range]) + $rnd_min ]
+        puts "\r(\$i/$repeat_count) $pcm $idx: Wait for \$sleep_t ms before resume"
+        send " "
+        after \$sleep_t
+        incr i
+        if { \$i > $repeat_count } { exit 0 }
+        exp_continue
+    }
+}
+exit 1
+END
+}
+
+max_wait_time=$[ 10 * $repeat_count ]
+
+for i in $(seq 1 $loop_count)
+do
+    dlogi "Loop count( $i / $loop_count )"
+    sudo dmesg -c >/dev/null
+    for pipeline_combine_str in "${pipeline_combine_lst[@]}"
+    do
+        unset pid_lst
+        declare -a pid_lst
+        for idx in $(echo $pipeline_combine_str)
+        do
+            func_pause_resume_pipeline $idx
+            pid_lst=(${pid_lst[*]} $!)
+        done
+        # wait for expect script finished
+        dlogi "wait for expect process finished"
+        i=0
+        while [ "$(pidof expect)" ]
+        do
+            i=$[ $i + 1 ]
+            sleep 1s
+            [[ $i -ge $max_wait_time ]] && break
+        done
+        if [ "$(pidof expect)" ]; then
+            dloge "Still have expect process not finished after wait for $max_wait_time"
+            # now dump process
+            ps -ef |grep -E 'aplay|arecord'
+            exit 1
+        fi
+        # now check for all expect quit status
+        # dump the pipeline combine, because pause resume will have too many operation log
+        for idx in $(echo $pipeline_combine_str)
+        do
+            pipeline_index=${pipeline_idx_lst[$idx]}
+            pcm=$(func_pipeline_parse_value $pipeline_index pcm)
+            dlogi "pipeline: $pcm with ${cmd_idx_lst[$idx]}"
+        done
+        dlogi "Check expect exit status"
+        for pid in ${pid_lst[*]}
+        do
+            wait $pid
+            [[ $? -ne 0 ]] && dloge "pause resume is exit status error" && exit 1
+        done
+    done
+    sof-kernel-log-check.sh 0
+    [[ $? -ne 0 ]] && dloge "Catch error from dmesg" && exit 1
+done
+
+sof-kernel-log-check.sh $KERNEL_LAST_LINE
+exit $?
```

Comment:
It's better to add a blank line at the end of the file to avoid the warning.

Reply:
ok